lib/simplecov/buildkite/profiles.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+
+module SimpleCov::Buildkite::Profiles
+  def self.run(*args)
+    IO.popen(args, &:read).tap do
+      $?.success? or fail("Command exited with status #{$?.exitstatus}: #{args.join(" ")}")
+    end.chomp
+  end
+
+  def self.git(*args)
+    run 'git',
+        *args
+  end
+
+  def self.git_diff_names(*args, diff_filter: '')
+    git('diff',
+        '--name-only',
+        "--diff-filter=#{diff_filter}",
+        *args).split "\n"
+  end
+
+  def self.git_short_commit(commit)
+    git 'rev-parse',
+        '--short',
+        commit
+  end
+
+  def self.git_merge_base(*refs)
+    git 'merge-base',
+        *refs
+  end
+
+  SimpleCov.profiles.define 'buildkite' do
+    STDERR.puts 'SimpleCov::Buildkite profile initialising...'
+    fail('Not running on Buildkite') unless ENV['BUILDKITE'] == 'true'
+
+    branch_name = ENV['BUILDKITE_BRANCH']
+
+    STDERR.puts "branch_name=#{branch_name}"
+
+    base_branch_name = (
+      ENV['BUILDKITE_PULL_REQUEST_BASE_BRANCH'] ||
+      ENV['BUILDKITE_PIPELINE_DEFAULT_BRANCH']
+    )
```

Comment:
This ends up using the blank string if `BUILDKITE_PULL_REQUEST_BASE_BRANCH` is an empty string, but present. Need to make it not do that.

Reply:
It's annoying to do without relying on activesupport, but this should work:

```ruby
    base_branch_name = ENV['BUILDKITE_PULL_REQUEST_BASE_BRANCH']
    if base_branch_name.nil? || base_branch_name.empty?
      base_branch_name = ENV['BUILDKITE_PIPELINE_DEFAULT_BRANCH']
    end
 ```

🙌

Reply:
Integrated! Testing now...

Reply:
Yep! It's working!